pkg/controller/config/config_controller.go

```diff
@@ -497,15 +498,31 @@ func (r *ReconcileConfig) applyCommunityResources(req reconcile.Request, cfg *op
 }
 
 func transformManifest(cfg *op.Config, m *mf.Manifest, addnTfrms ...mf.Transformer) (mf.Manifest, error) {
+	rbManifest := m.Filter(roleBinding)
+	rest := m.Filter(mf.Not(roleBinding))
 	tfs := []mf.Transformer{
 		mf.InjectOwner(cfg),
 		transform.InjectNamespaceConditional(flag.AnnotationPreserveNS, cfg.Spec.TargetNamespace),
-		transform.InjectNamespaceRoleBindingSubjects(cfg.Spec.TargetNamespace),
 		transform.InjectNamespaceCRDWebhookClientConfig(cfg.Spec.TargetNamespace),
 		transform.InjectDefaultSA(flag.DefaultSA),
 	}
+
 	tfs = append(tfs, addnTfrms...)
-	return m.Transform(tfs...)
+	rest, err := rest.Transform(tfs...)
+	if err != nil {
+		return *m, err
+	}
+
+	tfs = []mf.Transformer{
+		mf.InjectOwner(cfg),
+		transform.InjectNamespaceRoleBindingConditional(flag.AnnotationPreserveNS,
+			flag.AnnotationPreserveRBSubjectNS, cfg.Spec.TargetNamespace),
```

Comment:
can we add 
`transform.InjectNamespaceConditional(flag.AnnotationPreserveNS, cfg.Spec.TargetNamespace),` 
and `transform.InjectNamespaceRoleBindingConditional(flag.AnnotationPreserveRBSubjectNS, cfg.Spec.TargetNamespace),` 

and handle metadata Namespace and subject->Namespace separately in 2 different transformers

Reply:
Problem is that transform.InjectNamespaceConditional would transform the namespace of subject also because it calls manifestival transform.InjectNamespace which transforms subject namespace.

Reply:
https://github.com/manifestival/manifestival/blob/17489fb533562ae0e7c47c14a4ac9bfe55e8af8c/transform.go#L66

It update subject here